lib/google/gax/configuration.rb

```diff
@@ -112,15 +116,21 @@ def self.config? obj
       #     validation fails. Defaults to `true`.
       #
       def initialize show_warnings: true, &block
-        @show_warnings = show_warnings
         @values = {}
-        @defaults = {}
-        @validators = {}
+        @schema = Schema.new show_warnings: show_warnings
 
         # Can't call yield because of BasicObject
         block&.call self
       end
 
+      ##
+      # Derive a Configuration object. The subsequent object can set local values,
+      # but will not be able to change the structure.
```

Comment:
Can you document the block and explain its usage? (An example would probably be helpful too.)

Reply:
I think the best example would show how the block enables a "DSL"-type configuration style suitable for use in a Rails initializer, for example.

Reply:
I've updated the docs, but didn't add any code examples.